command/snapshot/inspect/snapshot_inspect_test.go

```diff
@@ -110,3 +133,48 @@ func TestSnapshotInspectCommand(t *testing.T) {
 		}
 	}
 }
+
+func TestSnapshotInspectEnhanceCommand(t *testing.T) {
+	t.Parallel()
+	a := agent.NewTestAgent(t, ``)
+	defer a.Shutdown()
+	client := a.Client()
+
+	dir := testutil.TempDir(t, "snapshot")
+	defer os.RemoveAll(dir)
+
+	file := filepath.Join(dir, "backup.tgz")
+
+	// Save a snapshot of the current Consul state
+	f, err := os.Create(file)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	snap, _, err := client.Snapshot().Save(nil)
+	if err != nil {
+		f.Close()
+		t.Fatalf("err: %v", err)
+	}
+	if _, err := io.Copy(f, snap); err != nil {
+		f.Close()
+		t.Fatalf("err: %v", err)
+	}
+	if err := f.Close(); err != nil {
+		t.Fatalf("err: %v", err)
+	}
```

Comment:
Do you still need all of this, or was this for debugging? I think you might be able to reduce this to:

```suggestion
	snap, _, err := client.Snapshot().Save(nil)
	require.NoError(t, err)
	defer snap.Close()
```

Reply:
Committed this but commenting here for further discussion tomorrow. 
I did this and passed it into `	args := []string{"-enhance", snap}`, since `file` was no longer a thing after this change, and because `snap` is an io.ReadCloser and not an `os.File` I was getting errors. (err: `annot use snap (type io.ReadCloser) as type string in slice literal`) 
Wondering if because os.File implements io.Reader that is why we can pass an os.File -> io.Reader but not reverse? 

Anyways reversed this code back for now :)

Reply:
Ah, that's right! I was thinking this was the return from our new function, but it's the return from the api package.